nodebuilder/state/core.go

```diff
@@ -20,6 +20,11 @@ func coreAccessor(
 	sync *sync.Syncer[*header.ExtendedHeader],
 	fraudServ libfraud.Service,
 ) (*state.CoreAccessor, *modfraud.ServiceBreaker[*state.CoreAccessor]) {
+	if !corecfg.EndpointConfigured() {
+		log.Info("No core endpoint provided, running node without state access")
+		return nil, nil
+	}
```

Comment:
We have a nice `fxutil.ProvideIf`. Why not it?

Reply:
less LOCs now: https://github.com/celestiaorg/celestia-node/pull/2577/commits/1b2435685721246bc28d3380da8453d9b28fb745